src/ActionModal/utils/MessageConstructor.js

```diff
@@ -62,174 +56,31 @@ export const getMultiMessage = async (context, messages) => {
   }
 }
 
-// Bank
 /* istanbul ignore next */
-export function MsgSend(
-  senderAddress,
-  {
-    toAddress,
-    amounts // [{ denom, amount}]
-  }
-) {
-  return {
-    type: `cosmos-sdk/MsgSend`,
-    value: {
-      from_address: senderAddress,
-      to_address: toAddress,
-      amount: amounts.map(Coin)
-    }
-  }
-}
+async function getMessageFormatter(network, messageType) {
+  let networkMessages
+  try {
+    networkMessages = await import(`./networkMessages/${network}.js`)
+  } catch (err) {
+    throw new Error("Signing for this network is not enabled.")
+  }
+  const messageFormatter = networkMessages[messageType]
 
-// Staking
-export function MsgDelegate(
-  senderAddress,
-  { validatorAddress, amount, denom }
-) {
-  /* istanbul ignore next */
-  return {
-    type: `cosmos-sdk/MsgDelegate`,
-    value: {
-      delegator_address: senderAddress,
-      validator_address: validatorAddress,
-      amount: Coin({ amount, denom })
-    }
+  if (!messageFormatter) {
+    throw new Error(
+      `Creating the message ${messageType} for the network ${network} is not supported`
```

Comment:
```suggestion
      `Creating the '${messageType}' message for the ${network} network is not supported`
```

Reply:
😸

Reply:
on second thought: these are user error messages so i think they should be helpful and actionable. maybe we could say "there's currently an issue with x transactions on this network. the lunie team is working on a fix. please try again later"

would that make sense?

Reply:
Same here: We should prevent this from ever surfacing to the user. The error is a backup